src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -26,9 +26,46 @@ class SymbolStore(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]
     assignSymbolStoreEntry(fullyQualifiedName, value, input = true)
   }
 
+  binding.workflows foreach {workflow =>
+    workflow.calls foreach {call =>
+      call.inputMappings foreach {
+        case (k, v) => assignSymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, input = true)
+      }
+    }
+  }
+
   def locallyQualifiedInputs(call: Call): Map[String, WdlValue] = {
+
+    def lookup(s: String): WdlValue = {
+      var workflow = call.parent.getOrElse {
+        throw new WdlExpressionException("Expecting 'call' to have a 'workflow' parent")
+      }.asInstanceOf[Workflow]
+      val matchedCall = workflow.calls.find{c => c.name == s}.getOrElse {
+        throw new WdlExpressionException(s"Expecting to find a call with name '$s'")
+      }
+      val callOutputs = store.collect{ case entry if entry.scope == matchedCall.fullyQualifiedName && entry.isOutput =>
+          val value = entry.wdlValue match {
+            case Some(v) => v
+            case _ => throw new WdlExpressionException(s"Could not evaluate call '${matchedCall.name}', because '${entry.key.name}' is undefined")
+          }
+          entry.key.name -> value
+      }.toMap[String, WdlValue]
+      WdlObject(callOutputs)
+    }
+
```

Comment:
There's definitely benefit to the customized error messages, and the following suggestion would lose that, so YMMV. This whole thing could be composed monadically a bit more, w/ something like this:

      val foo = call.parent.map {_.asInstanceOf[Workflow]}
      val matchedFoo = foo.map {_.calls.find {_.name == s}}
      val fooOutputs = ???

Where the last line was just me being too lazy to replicate the collect. But then you lose out on the 'why is this None, specifically?' angle.